Refactor FXIOS-10467 - Remove force_cast violations from Frontend Features & Content View

bmihai23 opened the pull request:
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-10467)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/22916)

## :bulb: Description
- This PR remove `force_cast` violations from: 
   - PocketViewModel
   - FindInPageHelper
   - LocalRequestHelper
   - ReaderPanel
   - DownloadsPanel
   - CertificatesViewController
- This PR is part of a series aimed at adding a new SwiftLint rule in the project, as described in #22916

## :pencil: Checklist
You have to check all boxes before merging
- [ ] Filled in the above information (tickets numbers and description of your work)
- [ ] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

cyndichin reviewed (approved):
looks great! thanks for making these changes! 🥳

cyndichin commented on firefox-ios/Client/Frontend/Home/Pocket/PocketViewModel.swift:
```diff
@@ -208,15 +208,19 @@ extension PocketViewModel: HomepageSectionHandler {
-                                                          for: indexPath) as! LegacyPocketStandardCell
+            guard let cell = collectionView.dequeueReusableCell(withReuseIdentifier: LegacyPocketStandardCell.cellIdentifier,
+                                                                for: indexPath) as? LegacyPocketStandardCell else {
+                fatalError("Failed to dequeue cell with identifier \(LegacyPocketStandardCell.cellIdentifier)")
```
this is an improvement to what we had before while also mimicking similar to what we had. If we want to fail more gracefully, we can also return a generic `UICollectionViewCell` with a fatal log error.

cyndichin commented on firefox-ios/Client/Frontend/Home/Pocket/PocketViewModel.swift:
```diff
@@ -208,15 +211,25 @@ extension PocketViewModel: HomepageSectionHandler {
+                                                                for: indexPath) as? LegacyPocketStandardCell else {
+                logger.log("Failed to dequeue LegacyPocketStandardCell at indexPath: \(indexPath)",
+                           level: .fatal,
+                           category: .homepage)
```
```suggestion
                           category: .legacyHomepage)
```
Since this is used in the legacy homepage, lets use the legacy category for now.

cyndichin commented on firefox-ios/Client/Frontend/Home/Pocket/PocketViewModel.swift:
```diff
@@ -208,15 +211,25 @@ extension PocketViewModel: HomepageSectionHandler {
+                                                                for: indexPath) as? PocketDiscoverCell else {
+                logger.log("Failed to dequeue PocketDiscoverCell at indexPath: \(indexPath)",
+                           level: .fatal,
+                           category: .homepage)
```
```suggestion
                           category: .legacyHomepage)
```

bmihai23 commented:
> thank you @bmihai23 for making the updates! appreciate the enhancement! added one minor thing to change, since we are currently working on rebuilding a new homepage and want to keep logging for the old vs. new homepage separate.

Hi @cyndichin, I just want to let you know that I updated the code based on your feedback, please let me know if it's okay

cyndichin commented:
> > thank you @bmihai23 for making the updates! appreciate the enhancement! added one minor thing to change, since we are currently working on rebuilding a new homepage and want to keep logging for the old vs. new homepage separate.
> 
> Hi @cyndichin, I just want to let you know that I updated the code based on your feedback, please let me know if it's okay

looks perfect! thank you so much!
